### PR TITLE
fix: fix performance RuntimeWarning

### PR DESCRIPTION
**Pull request by rickstaa**
This pull request ensures a warning is thrown instead of an exception. I think this is how pythons [warnings](https://docs.python.org/3/library/warnings.html) are meant to be used. It will now show the warning to developers while hiding it for end users who use your package as a module. Additionally, developers can turn these warnings into errors using the `-Werror` flag (see https://docs.python.org/3/using/cmdline.html#cmdoption-W). Feel free to move the `colorize` function wherever you want 👍🏻. If you want the warning to stop the code execution we should use [RuntimeError](https://docs.python.org/3/library/exceptions.html#RuntimeError) instead.

### CODE REVIEW

**Comment by rickstaa** [comment]
@jjshoots, maybe we can add [pre-commit](https://pre-commit.com/) to the (optional) dev dependencies and add this step to a CONTRIBUTING.md file in the future?

**Comment by rickstaa** [comment]
@jjshoots I just realised that for your package it is unnecessary to warp gym's colorize method as it can be used directly. I will fix this and remove the helpers module tomorrow morning 👍🏻.

File: PyFlyt/core/helpers.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        bold (bool, optional): Whether you want the text to be bold. Defaults to
+            ``False``.
+        highlight (bool, optional):  Whether you want to highlight the text. Defaults to
+            ``False``.
+
+    Returns:
+        str: Colorized string.
+    """
+    if color:  # If not empty.
+        return gym_colorize(string, color, bold, highlight)
```
Comment (jjshoots) [inline PyFlyt/core/helpers.py:24]: Is there any particular reason we don't just use this function directly?

**Comment by jjshoots** [comment]
Thanks for the fix! I'm not very familiar with the python warnings system I'm afraid, what's the difference between `RuntimeWarning` and using `warnings.warn`?

On the front of the warnings showing up for devs versus end users, I think what I'll do is to actually move the warning into the actual camera component and the render functions.
That way, it won't show up when the user doesn't actually need it.
That said, it's quite weird that pybullet doesn't have an option to automatically install with numpy build.

Also, yeah, adding a CONTRIBUTING.md is definitely on the to-do list, as well as a [dev] dependency that adds pytest and pre-commit. I just never expected anyone to be using it so quickly yet. 😅

PyFlyt also probably needs a proper release procedure, perhaps using the release GH action.
I'll try to address these tomorrow.

**Comment by jjshoots** [comment]
> Thanks for the fix! I'm not very familiar with the python warnings system I'm afraid, what's the difference between `RuntimeWarning` and using `warnings.warn`?
> 
> On the front of the warnings showing up for devs versus end users, I think what I'll do is to actually move the warning into the actual camera component and the render functions. That way, it won't show up when the user doesn't actually need it. That said, it's quite weird that pybullet doesn't have an option to automatically install with numpy build.
> 
> Also, yeah, adding a CONTRIBUTING.md is definitely on the to-do list, as well as a [dev] dependency that adds pytest and pre-commit. I just never expected anyone to be using it so quickly yet. sweat_smile
> 
> PyFlyt also probably needs a proper release procedure, perhaps using the release GH action. I'll try to address these tomorrow.

**Comment by jjshoots** [comment]
Whoops, didn't mean to close it.

**Comment by rickstaa** [comment]
> Thanks for the fix! I'm not very familiar with the python warnings system I'm afraid, what's the difference between RuntimeWarning and using warnings.warn?

The [warnings](https://docs.python.org/3/library/warnings.html) give more gradual control of the behaviour that happens when a problem is experienced. It provides a way to simply notify users without exiting the script. The [RuntimeWarning](https://docs.python.org/3/library/exceptions.html#RuntimeWarning) however inherits from the `Exception` class and can therefore be used to show an error and terminate the script as you do in the current codebase. However, when used with the warn method of the warning module it will show up in the console but without exiting the script. Developers can change this behaviour from outside your package. Which one to use `RuntimeError` (which exits the script) or `RuntimeWarning` in combination with the warnings module (which by default only shows a warning) depends on how severe the Runtime problem is. I wasn't aware of this module until some weeks ago when I found it used in the Gymnasium package 🤓.

> On the front of the warnings showing up for devs versus end users, I think what I'll do is actually move the warning into the actual camera component and the render functions.

This sounds like a good idea since in that case having the wrong Pybullet installed will result in problems. I think here you can even use a `RuntimeError` if you want to prevent people form using your package without Numpy.

> That said, it's quite weird that pybullet doesn't have an option to automatically install with numpy build.

Your right! I also don't like the way Pybullet handles this problem. Numpy should have been included as a build dependency there so that it is always built with Numpy. Alternatively, they should offer a `pybullet-numpy` version if they really want to keep support without numpy. I will create a feature request to try to solve this 👍🏻.

> Also, yeah, adding a CONTRIBUTING.md is definitely on the to-do list, as well as a [dev] dependency that adds pytest and pre-commit. I just never expected anyone to be using it so quickly yet. sweat_smile
>
> PyFlyt also probably needs a proper release procedure, perhaps using the release GH action.
> I'll try to address these tomorrow.
> Thanks for the help, though!

Ha, for some reason, your SEO is extremely good. I immediately got your article when I searched `Qaudrotor simulation` 🚀. No hurries, though, with the CONTRIBUTING.md as it is a low priority. I just mentioned it because you perform things in GitHub actions without enforcing them through pre-commit hooks on local systems. I think this confusion can already be solved by adding `pre-commit` as a dev dependency since a lot of developers will inspect the dev dependencies used before creating a pull request. Having that said, if you want an effortless release pipeline, you can check out https://github.com/googleapis/release-please. I use it in combination with https://github.com/google-github-actions/release-please-action, and have to say I'm very happy with it.

**Comment by rickstaa** [comment]
@jjshoots I removed the redundant helper's module. Feel free to merge it or close it based on where you want to handle this problem 👍🏻.

**Comment by rickstaa** [comment]
@jjshoots I created a discussion post on the bullet3 repository (see https://github.com/bulletphysics/bullet3/discussions/4505).

**Comment by jjshoots** [comment]
@rickstaa Awesome, thanks for the assist! I'm closing this in favour of: https://github.com/jjshoots/PyFlyt/commit/6aad763d2da618d82b78c841116c9274ae2a13ba 

It shouldn't throw errors in instances where the user doesn't need it anymore.

**Pull request closed without merging**